cou/utils/juju_utils.py

```diff
@@ -370,6 +370,7 @@ async def get_applications(self) -> dict[str, Application]:
                 workload_version=status.workload_version,
             )
             for app, status in full_status.applications.items()
+            if status.status.info != "waiting for machine"
```

Comment:
Like this we will skip whole app and I do not think it's necessary, since only single unit can wait for machine.